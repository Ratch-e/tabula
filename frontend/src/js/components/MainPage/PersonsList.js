/**
 * Список пользователей
 */
import * as React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as personsAction from '../../actions/PersonsActions';
import { BootstrapTable, TableHeaderColumn } from 'react-bootstrap-table';
import 'react-bootstrap-table/dist/react-bootstrap-table-all.min.css';
import 'bootstrap/dist/css/bootstrap.min.css';

class PersonsList extends React.Component {
  generatePersonList() {
    const list = this.props.persons.personList;
    let persons = list.map((person, key) => {
      let params = person.params || {};
      let catNames = Object.keys(params);
      return {
        id: key + 1,
        name: `${person.lastName} ${person.name}`,
        user_id: person._id,
        occupation: person.occupation,
        ...params,
        catNames: catNames,
      };
    });

    function linkFormatter(cell, row) {
      return `<a href="profile/${row.user_id}">${cell}</a>`;
    }

    const arr = [
      <TableHeaderColumn key="1" dataField="id" dataSort={true} width="50px">
        №
      </TableHeaderColumn>,

      <TableHeaderColumn key="2" dataField="name" dataSort={true} dataFormat={linkFormatter} width="150">
        Имя
      </TableHeaderColumn>,

      <TableHeaderColumn key="3" dataField="occupation" dataSort={true}>
        Должность
      </TableHeaderColumn>,
    ];

    if (persons.length) {
      for (const item in persons[0].catNames) {
        arr.push(
          <TableHeaderColumn key={item + 3} dataField={persons[0].catNames[item]} dataSort={true}>
            {persons[0].catNames[item]}
          </TableHeaderColumn>,
        );
      }
    }

    return persons.length ? (
      <BootstrapTable
        keyField="id"
        data={persons}
        search
        searchPlaceholder="Найти..."
        exportCSV
        csvFileName="Таблица_работников"
        ignoreSinglePage
        pagination
        striped
        hover
      >
        {arr}
      </BootstrapTable>
    ) : (
      0
    );
  }

  /**
   * Загрузка списка пользователей при загрузке компонента
   */
  componentDidMount() {
    this.props.personActions.fetchPersons();
  }

  render() {
    return <div className="person-list">{this.generatePersonList()}</div>;
  }
}

function mapStateToProps(state) {
  return {
    persons: state.persons,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    personActions: bindActionCreators(personsAction, dispatch),
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(PersonsList);
