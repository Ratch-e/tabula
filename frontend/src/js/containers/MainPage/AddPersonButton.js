/**
 * Кнопка создания нового пользователя
 */
import * as React from "react";

class AddPersonButton extends React.Component {
  render() {
    return (
        <a className="button" href="/new_profile">Добавить анкету</a>
    );
  }
}

export default AddPersonButton;