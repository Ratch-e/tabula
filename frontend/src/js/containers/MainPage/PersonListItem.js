import * as React from "react";

class PersonsListItem extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      name: props.name,
      lastname: props.lastName,
      id: props.id
    };
  }

  getFullname() {
    return `${this.state.lastname} ${this.state.name}`
  }

  render() {
    return (
        <a href={`/profile/${this.state.id}`}>{this.getFullname()}</a>
    );
  }
}

export default PersonsListItem;