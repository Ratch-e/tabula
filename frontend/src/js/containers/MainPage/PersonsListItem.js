import * as React from "react";

class PersonsListItem extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      name: props.name,
      lastname: props.lastName,
      id: props.id
    };
  }

  getFullname() {
    return `${this.state.lastname} ${this.state.name}`
  }

  render() {
    return (
      <div className="person-list__item">
        <a className="person-list__link" href={`/profile/${this.state.id}`}>       {this.getFullname()}
        </a>
      </div>
    );
  }
}

export default PersonsListItem;