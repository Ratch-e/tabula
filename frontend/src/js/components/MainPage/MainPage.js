/**
 * Главная страница
 */
import * as React from "react";
import Header from "../Header/Header"
import PersonsList from "../../containers/MainPage/PersonsList"

class MainPage extends React.Component {
  render() {
    return (
      <div className="page">
        <Header/>
        <div className="content main-page">
          <div className="main-page__actions">
            <a className="button" href="/new_profile">Добавить анкету</a>
            <a className="button" href="/occupations">Добавить профессию</a>
          </div>
          <PersonsList/>
        </div>
      </div>
    );
  }
}

export default MainPage;