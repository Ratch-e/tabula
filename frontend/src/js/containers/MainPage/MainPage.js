import * as React from "react";
import Header from "../Header/Header"
import AddPersonButton from "../../components/MainPage/AddPersonButton"


class MainPage extends React.Component {
  render() {
    return (
      <div className="page">
        <Header/>
        <div className="content">
            <AddPersonButton/>
        </div>
      </div>
    );
  }
}

export default MainPage;