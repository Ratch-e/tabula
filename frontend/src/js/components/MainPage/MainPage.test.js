import React from 'react';
import { shallow } from 'enzyme';
import MainPage from './MainPage';

describe('Главная страница', () => {
  it('рендерится корректно', () => {
    const component = shallow(<MainPage />);
    expect(component.length).toMatchSnapshot();
  });
});
