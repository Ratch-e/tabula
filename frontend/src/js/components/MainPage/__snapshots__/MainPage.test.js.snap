// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Главная страница рендерится корректно 1`] = `1`;
